oas3.yaml

```diff
@@ -0,0 +1,328 @@
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PetV1"
+        404:
+          description: The pet could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        500:
+          description: An error occurred on the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+    patch:
+      tags:
+        - pets
+      summary: Mofify Pet (v1)
+      description: Modifies the information about a pet
+      operationId: pet_modify_v1
+      parameters:
+        - name: id
+          in: path
+          description: The unique ID of the idea you wish to retrieve.
+          required: true
+          schema:
+            type: string
+            pattern: ^[a-f0-9]{32}$
+      requestBody:
+        description: The changes you wish to make to the pet's information.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PetV1"
+            example:
+              status: adopted
+
+      responses:
+        200:
+          description: The newly modified pet's information.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PetV1"
+        404:
+          description: The pet could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        500:
+          description: An error occurred on the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+components:
+  schemas:
+    HealthV1:
+      required:
+        - status
+      type: object
+      properties:
+        status:
+          type: string
+          description: The current aggregate status of the service.
+          enum:
+            - Healthy
+            - Degraded
+            - Unhealthy
+        results:
+          type: object
+          additionalProperties:
+            type: object
+            required:
+              - status
+              - description
+            properties:
+              status:
+                type: string
+                description: The current status of the service's component.
+                enum:
+                  - Healthy
+                  - Degraded
+                  - Unhealthy
+              description:
+                type: string
+                description: A short description of this component's healthy.
+              data:
+                type: object
+                additionalProperties: true
+                properties: {}
+          readOnly: true
+          xml:
+            name: Status
+      xml:
+        name: Health
+      example:
+        ok: true
+
+    PetV1:
+      required:
```

Comment:
Should status be required?

Reply:
In this case I think so, every upstream dependency should report a status of some kind otherwise it doesn't really have any business being present in the health report for the service (that's my $0.02 anyway).

Reply:
I agree.